modules/temp-channels/channel-settings.js

```diff
@@ -0,0 +1,282 @@
+            ]
+        }
+    });
+    const allowedUsersArray = vc.allowedUsers.split(',');
+    let allowedUsers = '';
+    for (const user of allowedUsersArray) {
+        allowedUsers = allowedUsers + '\n • <@' + user + '>';
+    }
+    if (allowedUsersArray.at(0) === '') {
+        interaction.editReply(localize('temp-channels', 'no-added-user'));
+        return;
+    }
+    interaction.editReply(moduleConfig['listUsers'] + ' ' + allowedUsers);
+};
+
+module.exports.channelEdit = async function (interaction, callerInfo) {
+    const moduleConfig = interaction.client.configurations['temp-channels']['config'];
+    const vc = await client.models['temp-channels']['TempChannel'].findOne({
+        where: {
+            [Op.and]: [
+                {id: interaction.member.voice.channelId},
+                {creatorID: interaction.member.id}
+            ]
+        }
+    });
+    const vchann = interaction.guild.channels.cache.get(vc.id);
+    let edited = 0;
+    let vcNsfw;
+    let vcBitrate;
+    let vcLimit;
+    let vcName;
+
+    if (callerInfo === 'command') {
+        if (interaction.options.getInteger('user-limit') >= 0) {
+            if (interaction.options.getInteger('user-limit') < 0 || interaction.options.getInteger('user-limit') > 99) {
+                interaction.editReply(localize('temp-channels', 'edit-error'));
+                return;
+            }
+            vcLimit = interaction.options.getInteger('user-limit');
+            edited++;
+        } else vcLimit = vchann.userLimit;
+        if (interaction.options.getInteger('bitrate')) {
+            if (interaction.options.getInteger('bitrate') <= 8000 || interaction.options.getInteger('bitrate') >= interaction.guild.maximumBitrate) {
+                interaction.editReply(localize('temp-channels', 'edit-error'));
+                return;
+            }
+            vcBitrate = interaction.options.getInteger('bitrate');
+            edited++;
+        } else vcBitrate = vchann.bitrate;
+        if (interaction.options.getString('name')) {
+            vcName = interaction.options.getString('name');
+            edited++;
+        } else vcName = vchann.name;
+        if (interaction.options.getBoolean('nsfw')) {
+            vcNsfw = interaction.options.getBoolean('nsfw');
+            edited++;
+        } else vcNsfw = vchann.nsfw;
+    }
+    if (callerInfo === 'modal') {
+        if (interaction.fields.getTextInputValue('edit-modal-limit-input') >= 0) {
+            if (interaction.fields.getTextInputValue('edit-modal-limit-input') < 0 || interaction.fields.getTextInputValue('edit-modal-limit-input') > 99) {
+                interaction.editReply(localize('temp-channels', 'edit-error'));
+                return;
+            }
+            vcLimit = interaction.fields.getTextInputValue('edit-modal-limit-input');
+            edited++;
+        } else vcLimit = vchann.userLimit;
+        if (interaction.fields.getTextInputValue('edit-modal-bitrate-input')) {
```

Comment:
Please make sure that bitrate is an actual number to prevent crashes

Reply:
Same for the limit please. Isn't there any way to only allow number entry in modals? That seems like a basic feature, considering that slash-commands support that type already

Reply:
No there isn't... there are only types for long and short inputs, nur for content types